custom/icds_reports/cache.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from quickcache import get_quickcache, ForceSkipCache
+from quickcache.django_quickcache import tiered_django_cache
+
+from corehq.sql_db.routers import forced_citus
+from corehq.util.quickcache import quickcache_soft_assert, get_session_key
+
+
+def get_cache_prefix_func(catch_force_skip=False):
+    def prefix_func():
+        try:
+            prefix = get_session_key()
+        except ForceSkipCache:
```

Comment:
Why do we need this part?

Reply:
The session ID function is normally only used in the memory cache and raises exception if the session ID is empty (anywhere outside a request or celery task). Since we are using it for the redis cache as well we need to ignore that exception for that cache.

Reply:
makes sense

Reply:
Does that mean caching will now only be per user?

Reply:
actually, it looks like that value is set per request. Does this prevent caching across requests? Either of those things could be a pretty significant performance hit